pkg/controller/sesv2/configurationset/setup.go

```diff
@@ -0,0 +1,305 @@
+		cr.Status.SetConditions(xpv1.Creating())
+	}
+
+	// Passing ConfigurationSet object from Observation into hooks for Update function to access
+	e.ConfigurationSetObservation = resp
+
+	return obs, nil
+}
+
+func preCreate(_ context.Context, cr *svcapitypes.ConfigurationSet, obj *svcsdk.CreateConfigurationSetInput) error {
+	obj.ConfigurationSetName = awsclient.String(meta.GetExternalName(cr))
+	return nil
+}
+
+func preDelete(_ context.Context, cr *svcapitypes.ConfigurationSet, obj *svcsdk.DeleteConfigurationSetInput) (bool, error) {
+	obj.ConfigurationSetName = awsclient.String(meta.GetExternalName(cr))
+	return false, nil
+}
+
+func (e *hooks) update(ctx context.Context, mg resource.Managed) (managed.ExternalUpdate, error) { // nolint:gocyclo
+	cr, ok := mg.(*svcapitypes.ConfigurationSet)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errUnexpectedObject)
+	}
+	// Update Resource is not provided other than individual PUT operation
+	// NOTE: Update operation NOT generated by ACK code-generator
+
+	// Populate ConfigurationSetName from meta.AnnotationKeyExternalName
+	configurationSetName := awsclient.String(mg.GetAnnotations()[meta.AnnotationKeyExternalName])
+
+	// update DeliveryOptions (PutConfigurationSetDeliveryOptions)
+	if !isUpToDateDeliveryOptions(cr, e.ConfigurationSetObservation) {
+		deliveryOptionsInput := &svcsdk.PutConfigurationSetDeliveryOptionsInput{
+			ConfigurationSetName: configurationSetName,
+			SendingPoolName:      cr.Spec.ForProvider.DeliveryOptions.SendingPoolName,
+			TlsPolicy:            cr.Spec.ForProvider.DeliveryOptions.TLSPolicy,
+		}
+		if _, err := e.client.PutConfigurationSetDeliveryOptionsWithContext(ctx, deliveryOptionsInput); err != nil {
+			return managed.ExternalUpdate{}, errors.Wrap(err, "update failed")
+		}
+	}
+
+	// update ReputationOptions (PutConfigurationSetReputationOptions)
+	if !isUpToDateReputationOptions(cr, e.ConfigurationSetObservation) {
+		reputationOptionsInput := &svcsdk.PutConfigurationSetReputationOptionsInput{
+			ConfigurationSetName:     configurationSetName,
+			ReputationMetricsEnabled: cr.Spec.ForProvider.ReputationOptions.ReputationMetricsEnabled,
+		}
+		if _, err := e.client.PutConfigurationSetReputationOptionsWithContext(ctx, reputationOptionsInput); err != nil {
+			return managed.ExternalUpdate{}, errors.Wrap(err, "update failed")
+		}
+	}
+
+	// update SuppressionOptions (PutConfigurationSetSuppressionOptions)
+	var suppresssedReasons []*string
+	if !isUpToDateSuppressionOptions(cr, e.ConfigurationSetObservation) {
+		suppresssedReasons = cr.Spec.ForProvider.SuppressionOptions.SuppressedReasons
+		supressOptionsInput := &svcsdk.PutConfigurationSetSuppressionOptionsInput{
+			ConfigurationSetName: configurationSetName,
+			SuppressedReasons:    suppresssedReasons,
+		}
+		if _, err := e.client.PutConfigurationSetSuppressionOptionsWithContext(ctx, supressOptionsInput); err != nil {
+			return managed.ExternalUpdate{}, errors.Wrap(err, "update failed")
```

Comment:
There should be a different error message for each `Wrap` to help identify where the error came from. `"update failed"` is too generic here.